crates/ide_completion/src/completions/record.rs

```diff
@@ -14,12 +14,31 @@ pub(crate) fn complete_record(acc: &mut Completions, ctx: &CompletionContext) ->
             | ImmediateLocation::RecordExprUpdate(record_expr),
         ) => {
             let ty = ctx.sema.type_of_expr(&Expr::RecordExpr(record_expr.clone()));
+
             let default_trait = ctx.famous_defs().core_default_Default();
-            let impl_default_trait = default_trait.zip(ty).map_or(false, |(default_trait, ty)| {
-                ty.original.impls_trait(ctx.db, default_trait, &[])
-            });
+            let impl_default_trait =
+                default_trait.zip(ty.as_ref()).map_or(false, |(default_trait, ty)| {
+                    ty.original.impls_trait(ctx.db, default_trait, &[])
+                });
 
-            let missing_fields = ctx.sema.record_literal_missing_fields(record_expr);
+            let missing_fields = match ty.and_then(|t| t.adjusted().as_adt()) {
```

Comment:
```suggestion
            let missing_fields = match ty.and_then(|t| t.original().as_adt()) {
```
shouldn't really matter but we want the unadjusted type of the record expression here, since we don't care about what is happening to the type of the expression here